integration_test/cases/info_test.exs

```diff
@@ -1,5 +1,60 @@
-defmodule TestInfo do
+defmodule InfoTest do
   use ExUnit.Case, async: true
+  alias TestPool, as: P
+  alias TestAgent, as: A
 
-  test "handle_info returns new state"
+  test "handle_info handles message and moves on" do
+    stack = [
+      fn opts ->
+        send(opts[:parent], {:connected, self()})
+        {:ok, :state}
+      end,
+      :ok,
+      {:idle, :state},
+      {:idle, :state}
+    ]
+
+    {:ok, agent} = A.start_link(stack)
+    {:ok, pool} = P.start_link(agent: agent, parent: self())
+
+    assert_receive {:connected, conn}
+    send(conn, "some harmless message")
+    assert P.run(pool, fn _ -> :result end) == :result
+
+    assert [
+             connect: _,
+             handle_info: _,
+             handle_status: _,
+             handle_status: _
+           ] = A.record(agent)
+  end
+
+  test "handle_info can force disconnect" do
```

Comment:
I still need to add a test where the message arrives when the client has checked out the connection.